src/netbase.cpp

```diff
@@ -235,8 +232,9 @@ CService LookupNumeric(const std::string& name, uint16_t portDefault, DNSLookupF
     CService addr;
     // "1.2:345" will fail to resolve the ip, but will still set the port.
     // If the ip fails to resolve, re-init the result.
-    if(!Lookup(name, addr, portDefault, false, dns_lookup_function))
-        addr = CService();
+    const std::optional<CService> addr_opt{Lookup(name, portDefault, false, dns_lookup_function)};
+    if (addr_opt.has_value())
+        addr = addr_opt.value();
```

Comment:
This can be simplified:

<details>
<summary>git diff</summary>

```diff
diff --git a/src/netbase.cpp b/src/netbase.cpp
index 534dafdb4..2d4341a4c 100644
--- a/src/netbase.cpp
+++ b/src/netbase.cpp
@@ -229,13 +229,9 @@ CService LookupNumeric(const std::string& name, uint16_t portDefault, DNSLookupF
     if (!ContainsNoNUL(name)) {
         return {};
     }
-    CService addr;
     // "1.2:345" will fail to resolve the ip, but will still set the port.
     // If the ip fails to resolve, re-init the result.
-    const std::optional<CService> addr_opt{Lookup(name, portDefault, false, dns_lookup_function)};
-    if (addr_opt.has_value())
-        addr = addr_opt.value();
-    return addr;
+    return Lookup(name, portDefault, false, dns_lookup_function).value_or(CService{});
 }
 
 /** SOCKS version */
```
</details>

Reply:
Great, gonna implement it! Simpler!